server/backends/chunkstore/chunkstore.go

```diff
@@ -577,8 +582,12 @@ func (l *writeLoop) run(ctx context.Context) {
 		l.writeResultChannel <- result
 
 		if l.writeHook != nil {
-			// All work has been done for this write; trigger the writeHook
-			l.writeHook(ctx, req, result, chunk, volatileTail)
+			if bytesFlushed != 0 || !open || (req != nil && ((req.Chunk != nil && len(req.Chunk) != 0) || req.VolatileTail != nil)) {
```

Comment:
nit: might be able to simplify this by having a method like `req.IsEmpty()`

Reply:
Done!